plugin/trino-password-authenticators/src/test/java/io/trino/plugin/password/ldap/LdapUtil.java

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.password.ldap;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.BasicAttribute;
+import javax.naming.directory.BasicAttributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.ModificationItem;
+
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.Character.MAX_RADIX;
+import static java.lang.Math.abs;
+import static java.lang.Math.min;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+final class LdapUtil
+{
+    private static final SecureRandom random = new SecureRandom();
+    private static final int RANDOM_SUFFIX_LENGTH = 10;
+
+    public static final String MEMBER = "member";
+
+    private LdapUtil()
+    {}
+
+    public static String addLdapDefinition(LdapObjectDefinition ldapObjectDefinition, DirContext context)
+    {
+        requireNonNull(ldapObjectDefinition, "LDAP Object Definition is null");
+
+        Attributes entries = new BasicAttributes();
+        Attribute objectClass = new BasicAttribute("objectClass");
+
+        ldapObjectDefinition.getAttributes()
+                .forEach((key, value) -> entries.put(new BasicAttribute(key, value)));
+
+        ldapObjectDefinition.getObjectClasses()
+                .forEach(objectClass::add);
+        entries.put(objectClass);
+
+        try {
+            context.createSubcontext(ldapObjectDefinition.getDistinguishedName(), entries);
+        }
+        catch (NamingException e) {
+            throw new RuntimeException("LDAP Entry addition failed", e);
+        }
+
+        return ldapObjectDefinition.getDistinguishedName();
+    }
+
+    public static void addAttributesToExistingLdapObjects(String distinguishedName, Map<String, List<String>> modifiedAttributes, DirContext context)
+    {
+        requireNonNull(distinguishedName, "distinguishedName is null");
+        requireNonNull(modifiedAttributes, "modifiedAttributes is null");
+
+        List<ModificationItem> modificationItems = modifiedAttributes.entrySet().stream()
+                .flatMap(entry -> entry.getValue().stream()
+                        .map(attribute -> new ModificationItem(DirContext.ADD_ATTRIBUTE, new BasicAttribute(entry.getKey(), attribute))))
+                .collect(toList());
+
+        try {
+            context.modifyAttributes(distinguishedName, modificationItems.toArray(ModificationItem[]::new));
```

Comment:
I actually meant this :)

But I was a little reluctant to make suggestions in this class because this is almost-copy-pasted code, though.

```suggestion
        ModificationItem[] modificationItems = modifiedAttributes.entrySet().stream()
                .flatMap(entry -> entry.getValue().stream()
                        .map(attribute -> new ModificationItem(DirContext.ADD_ATTRIBUTE, new BasicAttribute(entry.getKey(), attribute))))
                .toArray(ModificationItem[]::new);

        try {
            context.modifyAttributes(distinguishedName, modificationItems);
```

Reply:
It is kinda copy pasted from `Tempto` but I guess we could have this change.